src/main.ts

```diff
@@ -3,10 +3,12 @@ import * as exec from '@actions/exec'
 
 async function run() {
     try {
-        // await exec.exec('julia', ['--color=yes', '-e', 'using Pkg; Pkg.add("Coverage"); using Coverage; Coveralls.submit(process_folder())'])
-        await exec.exec('julia', ['--color=yes', '-e', 'using Pkg; Pkg.add(PackageSpec(name="Coverage", rev="master")); using Coverage; Coveralls.submit(process_folder())'])
+        await exec.exec('julia', ['--color=yes', '-e', 'using Pkg; Pkg.add("Coverage"); using Coverage; Coveralls.submit(process_folder())'])
     } catch (error) {
-        core.setFailed(error.message)
+        // We are making errors non fatal for now because this for example
+        // will generally not work if triggered by a PR, which is a common
+        // situation.
+        // core.setFailed(error.message)
```

Comment:
Perhaps this could be the decision of the user by setting [`jobs.<job_id>.steps.continue-on-error`](https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error) in the workflow?

If not, let's at least log the error message to allow debugging if the failure is unexpected:

```suggestion
        // core.setFailed(error.message)
        core.debug(error.message)
```

Reply:
I think a debug message might be best for now. Hopefully in some future version we'll be able to work around this problem, so maybe introducing a config now is not such a good idea.

Another idea I recently had is that maybe we shouldn't even handle the upload at all in the julia action... There is a coveralls action that uploads, I believe, maybe we should just use that instead?

In that design the test action might actually do the processing of the coverage file so that a `lcov` file gets generated, and then from there on the coveralls/codecov action handles the rest.

But not really sure...

Reply:
Sounds like a good idea, but I'm not sure what the best approach is.

Reply:
I just looked at the codecov action, and I think it might not really be in a good state to be used right now :) I uses a VM, which slows things down a lot. So maybe we just ignore that whole option for now.